tests/components/command_line/test_switch.py

```diff
@@ -2,209 +2,240 @@
-            test_switch = {
-                "command_state": f"cat {path}",
-                "command_on": f"echo 1 > {path}",
-                "command_off": f"echo 0 > {path}",
-            }
-            assert setup_component(
-                self.hass,
-                switch.DOMAIN,
-                {
-                    "switch": {
-                        "platform": "command_line",
-                        "switches": {"test": test_switch},
-                    }
-                },
-            )
-            self.hass.block_till_done()
-            state = self.hass.states.get("switch.test")
-            assert STATE_OFF == state.state
-
-            common.turn_on(self.hass, "switch.test")
-            self.hass.block_till_done()
-
-            state = self.hass.states.get("switch.test")
-            assert STATE_ON == state.state
-
-            common.turn_off(self.hass, "switch.test")
-            self.hass.block_till_done()
-
-            state = self.hass.states.get("switch.test")
-            assert STATE_ON == state.state
-
-    def test_assumed_state_should_be_true_if_command_state_is_none(self):
-        """Test with state value."""
-        # args: hass, device_name, friendly_name, command_on, command_off,
-        #       command_state, value_template
-        init_args = [
-            self.hass,
-            "test_device_name",
-            "Test friendly name!",
-            "echo 'on command'",
-            "echo 'off command'",
-            None,
-            None,
-            15,
-        ]
-
-        no_state_device = command_line.CommandSwitch(*init_args)
-        assert no_state_device.assumed_state
-
-        # Set state command
-        init_args[-3] = "cat {}"
-
-        state_device = command_line.CommandSwitch(*init_args)
-        assert not state_device.assumed_state
-
-    def test_entity_id_set_correctly(self):
-        """Test that entity_id is set correctly from object_id."""
-        init_args = [
-            self.hass,
-            "test_device_name",
-            "Test friendly name!",
-            "echo 'on command'",
-            "echo 'off command'",
-            False,
-            None,
-            15,
-        ]
-
-        test_switch = command_line.CommandSwitch(*init_args)
-        assert test_switch.entity_id == "switch.test_device_name"
-        assert test_switch.name == "Test friendly name!"
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.setup import async_setup_component
+
+
+async def test_state_none(hass):
+    """Test with none state."""
+    with tempfile.TemporaryDirectory() as tempdirname:
+        path = os.path.join(tempdirname, "switch_status")
+        test_switch = {
+            "command_on": f"echo 1 > {path}",
+            "command_off": f"echo 0 > {path}",
+        }
+        assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6259301)